x/lockup/types/genesis.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"fmt"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/althea-net/althea-chain/types"
+)
+
+// DefaultGenesisState creates a simple GenesisState suitible for testing
+func DefaultGenesisState() *GenesisState {
+	return &GenesisState{
+		Locked: true,
+		LockExempt: []string{
+			"0x0000000000000000000000000000000000000000",
+		},
+		LockedMessageTypes: []string{
+			banktypes.TypeMsgSend,
+			banktypes.TypeMsgMultiSend,
+		},
+	}
+}
+
+func (s GenesisState) ValidateBasic() error {
```

Comment:
Here's a point of confusion, do I need to implement ValidateBasic like this, or should I be using ParamSetPairs below? Not fully certain what's going on with gravity's Params struct and the paramSpace in the keeper. I'd like to have a conversation about this.

Reply:
you need both, param set pairs is a bit of boilterplate that registers paramater names to store values which can be updated by governance, without that set pairs call you'll be able to load, read, write, and use the parameter from within the code but not using a governance vote.

We also need to validate the parameters coming out of the genesis file and reject invalid ones at start (not to mention reject them if somehow they are set by governance) hence the validation strucure you see over in the gravity module.